### PR TITLE
Initialize utility token

### LINKED ISSUE #2: Initialize Utility Token Repository
This issue is blocked by #1 

UtilityToken contracts are used in more than one repositories. 
All repositories have its own implementations, so
- Its code duplication
- It is difficult to maintain code across the repositories.

Add/Move the following contracts in this repository
- UtilityToken.sol
- UtilityTokenInterface.sol
- UtilityTokenRequiredInterface.sol

## TODOs
- [ ] Add/move the contracts
- [ ] Add/Move the unit tests for all the above contracts
- [ ] Create README (look at the openst-contracts README; link to developer-guidelines)

### PR DESCRIPTION
**Pull request by sarvesh-ost**
fixes #2  

This PR initializes this rep: 
- [X] Copied contracts from mosaic-contracts.
- [X] Copied unit test cases
- [X] Updated truffle config
- [X] Added package.json
- [X] Added README.md

Blocked by #3

### CODE REVIEW

File: contracts/truffle/Migrations.sol [hunk 1]
```diff
@@ -0,0 +1,23 @@
+pragma solidity >=0.4.21 <0.6.0;
```
[the remark this answers is not in the document]
Comment (sarvesh-ost) [reply contracts/truffle/Migrations.sol]: This is an auto-generated contract with `truffle init` command. Now it's possible to specify the range in pragma statement. This statement means this contract can compile with all the versions between 0.4.21 and 0.6.0. 

I am not sure if we should change this 🤔 
Should this be changed?

File: contracts/SafeMath.sol [hunk 2]
```diff
@@ -0,0 +1,143 @@
+pragma solidity ^0.5.0;
```
[the remark this answers is not in the document]
Comment (sarvesh-ost) [reply contracts/SafeMath.sol]: Yes, agree. 
That will be done in a separate ticket.

File: README.md [hunk 3]
```diff
@@ -1 +1,77 @@
...
+
+## Compiling the contracts
+
+The following npm script compiles updated contracts from the last call:
+
+```bash
+npm run compile
+```
+
+, to compile all contracts, run:
```
[the remark this answers is not in the document]
Comment (sarvesh-ost) [reply README.md]: Done 👍

**Comment by sarvesh-ost** [comment]
> Nice work 👍
> Few inline comments.

Thanks for the review. There are some feedbacks which is about fixing documentation and formatting. 
I think that is outside the scope of this ticket, as this ticket is about moving contracts. 

What do you think?

File: contracts/truffle/Migrations.sol (hunk 1 above)
[the remark this answers is not in the document]
Comment (gulshanvasnani) [reply contracts/truffle/Migrations.sol]: If it is auto-generated then no need to change.

> > Nice work 👍
> > Few inline comments.
> 
> Thanks for the review. There are some feedbacks which is about fixing documentation and formatting.
> I think that is outside the scope of this ticket, as this ticket is about moving contracts.
> 
> What do you think?

I am fine to address formatting and fixing documentation in a separate ticket.

**Comment by gulshanvasnani** [comment]
> > Nice work 👍
> > Few inline comments.
> 
> Thanks for the review. There are some feedbacks which is about fixing documentation and formatting.
> I think that is outside the scope of this ticket, as this ticket is about moving contracts.
> 
> What do you think?

I am fine to address formatting and fixing documentation in a separate ticket.

File: contracts/truffle/Migrations.sol (hunk 1 above)
[the remark this answers is not in the document]
Comment (gulshanvasnani) [reply contracts/truffle/Migrations.sol]: As it is auto-generated then no need to change.

**Comment by sarvesh-ost** [comment]
> > > Nice work 👍
> > > Few inline comments.
> > 
> > 
> > Thanks for the review. There are some feedbacks which is about fixing documentation and formatting.
> > I think that is outside the scope of this ticket, as this ticket is about moving contracts.
> > What do you think?
> 
> I am fine to address formatting and fixing documentation in a separate ticket.
> Please create the ticket.

It seems creating a ticket was more effort than making changes. I have done all the changes related to formating.

**Pull request merged**